js/DataService.js

```diff
@@ -288,16 +287,25 @@ angular.module('PathOfDamage')
 
   function decodeElements(encodedElements) {
     if (encodedElements) {
-      var elements = [];
-      var booleans = parseInt(encodedElements, 36).toString(2).split('');
-      var index = booleans.length;
-      for (var i = ELEMENTS.length; i-- > 0;) {
-        elements[ELEMENTS[i]] = decodeBoolean(booleans[--index]);
-      }
+      var booleans = decodeBooleans(encodedElements, ELEMENTS.length);
+      var elements = {};
+      ELEMENTS.forEach(function (element, index) {
+        elements[element] = booleans[index];
+      });
```

Comment:
I couldn't find any convenient way to do this transformation, as I can't map since I'm going from an array to an object with named indices.